src/modules/UI/components/PasswordRecoveryModal/PasswordRecoveryModalActions.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+
+import { bns } from 'biggystring'
+import { Actions } from 'react-native-router-flux'
+
+import { RECOVER_PASSWORD } from '../../../../constants/indexConstants.js'
+import { getSyncedSettingsAsync, setSyncedSettingsAsync } from '../../../Core/Account/settings.js'
+import type { Dispatch, GetState } from '../../../ReduxTypes.js'
+import { tallyUpTotalCrypto } from '../../../utils.js'
+import { updateSettings as updateReduxSettings } from '../../Settings/action.js'
+
+export const SHOW_PASSWORD_RECOVERY_MODAL = 'SHOW_PASSWORD_RECOVERY_MODAL'
+export const HIDE_PASSWORD_RECOVERY_MODAL = 'HIDE_PASSWORD_RECOVERY_MODAL'
+
+export const checkPasswordRecovery = () => async (dispatch: Dispatch, getState: GetState) => {
+  const state = getState()
+  const settings = state.ui.settings
+  const account = state.core.account
+  const passwordRecoveryRemindersShown = { ...settings.passwordRecoveryRemindersShown }
```

Comment:
check if password recovery is setup,
if it is already setup, stop working as soon as possible

all code after that point should assume that password recovery is not setup